src/pymoca/tree.py

```diff
@@ -388,6 +418,15 @@ def _find_simple_name(
     # unless predefined type, function, operator. If name matches a variable (a.k.a. component,
     # symbol) in an enclosing class, it must be a `constant`.
 
+    # Step 0.1: Predefined types
+    if name in InstanceTree.BUILTIN_TYPES:
+        return scope.root.classes[name]
+
+    # Step 0.2: The scope itself
+    if scope.name == name:
```

Comment:
Lines 418-419 are *meant* to be a performance improvement (though not measured 😆), but also helps ensure that the spec is met because if someone stuck a `Real` class between here and root, I don't know if anything else in the current code would stop it from being found. Really, the parser should check for reserved names, but I don't remember there being one.

TODO: Add a TODO or issue to add a parser reserved name check (if not there already)

Line 422 that your comment appears to be on is sort of an edge case that came up in testing.